configuration_broker/config_broker.cc

```diff
@@ -0,0 +1,220 @@
+// Copyright Microsoft and CHERIoT Contributors.
+// SPDX-License-Identifier: MIT
+
+// Contributed by Configured Things Ltd
+
+#include "cdefs.h"
+#include "cheri.hh"
+#include <compartment.h>
+#include <cstdlib>
+#include <debug.hh>
+#include <fail-simulator-on-error.h>
+#include <futex.h>
+#include <thread.h>
+
+#include <string.h>
+#include <vector>
+
+#include "config_broker.h"
+#include "sandbox.h"
+
+// Import some useful things from the CHERI namespace.
+using namespace CHERI;
+
+// Expose debugging features unconditionally for this compartment.
+using Debug = ConditionalDebug<DEBUG_CONFIG_BROKER, "Config Broker">;
+
+// Internal represention of Configurtaion Items
+struct NamedConfigItem
+{
+	char      *name; // name matched to capabilities
+	ConfigItem item;
+};
+
+//
+// Set of config data items.
+//
+std::vector<struct NamedConfigItem *> configData;
```

Comment:
This could do with a comment.  You need this to be full of `NamedConfigItem*`s not `NamedConfigItem`s so that their locations are stable.  It would probably be better to have a list here because `std::vector` does not handle allocation failure well, whereas just pushing a new list item would work well.

```suggestion
std::vector<NamedConfigItem *> configData;
```

Reply:
Is std::list  implemented ?  I was that std::vector was  and has its own include file (probably saw that in the exercise), but can't see the same for list.   I seem to remember reading (or maybe hearing) that not all classes were implemented ?

As its a mapping from a name std::map migth be even better ?

Reply:
No, we have a `ds::list` class for building intrusive lists.